gateway/handlers/reader.go

```diff
@@ -6,20 +6,20 @@ package handlers
 import (
 	"context"
 	"encoding/json"
-	"fmt"
+	"log"
 	"net/http"
 
 	"strings"
 
-	"github.com/openfaas/faas/gateway/metrics"
-	"github.com/openfaas/faas/gateway/requests"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/filters"
 	"github.com/docker/docker/client"
+	"github.com/openfaas/faas/gateway/metrics"
+	"github.com/openfaas/faas/gateway/requests"
 )
 
 // MakeFunctionReader gives a summary of Function structs with Docker service stats overlaid with Prometheus counters.
-func MakeFunctionReader(metricsOptions metrics.MetricOptions, c *client.Client) http.HandlerFunc {
+func MakeFunctionReader(metricsOptions metrics.MetricOptions, c client.ServiceAPIClient) http.HandlerFunc {
```

Comment:
What changed RE: client.ServiceAPIClient?

Reply:
Using `client.ServiceAPIClient` allowed me to mock the call to `c.ServiceList` in the unit tests.

Reply:
Is that an interface in the Docker code?

Reply:
Yep https://github.com/openfaas/faas/blob/master/gateway/vendor/github.com/docker/docker/client/interface.go#L124